libs/ledger-live-common/src/bot/engine.ts

```diff
@@ -228,9 +226,20 @@ export async function runWithAppSpec<T extends Transaction>(
           account,
           accounts,
           mutationsCount,
-          syncAllAccountsTime,
+          resyncAccountsDuration,
+          accountIdsNeedResync,
           preloadedData,
         });
+        if (report.finalAccount) {
+          // optim: no need to resync if all went well with finalAccount
+          const finalAccount: Account = report.finalAccount;
+          accountIdsNeedResync = accountIdsNeedResync.filter(
+            (id) => id !== finalAccount.id
+          );
+          accounts = accounts.map((a: Account) =>
+            a.id === finalAccount.id ? finalAccount : a
```

Comment:
I don't get the need for this, what the difference between `a` and `finalAccount` here when `a.id === finalAccount.id`?

Reply:
a = account state before transaction
finalAccount = new account state after re-sync